### ISSUE TITLE
PHP 7.2 compatibility issue if Twig report has no rows

### ISSUE THREAD

**Issue by rungta** [barrelstrength/craft-sprout-reports#54]
This line is resulting in an error if the report’s results are a null set (basically `craft.sproutReports.addRow()` is never called). 

https://github.com/barrelstrength/craft-sprout-reports/blob/e3cf9b25c17cb8df1d7655c4df8ebb58f2ca318b/src/datasources/CustomTwigTemplate.php#L81

```
yii\base\ErrorException: count(): Parameter must be an array or an object that implements Countable in .../vendor/barrelstrength/sprout-reports/src/datasources/CustomTwigTemplate.php:81

Stack trace:
#0 .../vendor/craftcms/cms/src/web/ErrorHandler.php(81): yii\base\ErrorHandler->handleError(2, 'count(): Parame...', '/home/runcloud/...', 81)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'count(): Parame...', '/home/runcloud/...', 81, Array)
#2 .../vendor/barrelstrength/sprout-reports/src/datasources/CustomTwigTemplate.php(81): count(NULL)
#3 .../vendor/barrelstrength/sprout-base-reports/src/controllers/ReportsController.php(138): barrelstrength\sproutreports\datasources\CustomTwigTemplate->getResults(Object(barrelstrength\sproutbasereports\elements\Report))
#4 [internal function]: barrelstrength\sproutbasereports\controllers\ReportsController->actionResultsIndexTemplate('sprout-reports', Object(barrelstrength\sproutbasereports\elements\Report), 17763)
#5 .../vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#6 .../vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#7 .../vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('results-index-t...', Array)
#8 .../vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('results-index-t...', Array)
#9 .../vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('sprout-base-rep...', Array)
#10 .../vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('sprout-base-rep...', Array)
#11 .../vendor/craftcms/cms/src/web/Application.php(286): yii\web\Application->handleRequest(Object(craft\web\Request))
#12 .../vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#13 .../web/index.php(21): yii\base\Application->run()
#14 {main}
```

I have a feeling this is similar to #29 and https://github.com/barrelstrength/craft-sprout-reports/commit/60210c8d9664684ba1a3ffca35849e3215842518 because I’m not getting the error in a PHP 7.1.x environment but getting the error in a PHP 7.2.x environment.

**Comment by BenParizek** [comment]
Thanks @rungta We'll look into it and update this thread when we have a solution.

**Comment by andrelopez** [comment]
Hi @rungta we just released Sprout Reports v1.1.7 with a fix to this issue.

**Comment by rungta** [comment]
Thanks! Works as expected now.
